### PR TITLE
Add geometry raytrace test app

### PR DESCRIPTION
**Pull request by sethrj**
This version is a little hacky:
- Uses JSON + raw binary output instead of something sensible
- Rendering the image only easy with "export controlled" software (supplementary code in SCALE that I plan to release as a standalone package but for now is private to me/tom/stefano)

### CODE REVIEW

**Comment by sethrj** [comment]
Closes #22

**Comment by sethrj** [comment]
Rendered images of CMS:

cms-plot
cms-render

with input:
```json
{"image":
    {"lower_left": [-500,0,-1000], "upper_right": [500,0,1000],
     "rightward_ax": [0,0,1],
     "vertical_pixels": 4096},
 "cuda_stack_size": 32768,
 "input": "../test/geometry/data/cms2018.gdml",
"output": "cms.bin"}
```

File: src/geometry/GeoParams.cc [hunk 1]
```diff
@@ -125,5 +139,19 @@ GeoParamsPointers GeoParams::device_pointers() const
     return result;
 }
 
+//---------------------------------------------------------------------------//
+/*!
+ * Increase CUDA stack size to enable complex geometries.
+ *
+ * For example, CMS requires a stack limit of at least 8192 * 4.
```
Comment (pcanal) [inline src/geometry/GeoParams.cc:146]: The actual requirement could be less, I did not test.  From what I saw 8192 * 2 might work.

Comment (sethrj) [reply src/geometry/GeoParams.cc:146]: OK, so maybe change wording to:
> For the `cms2018.gdml` detector geometry, the default stack size is too small, and a limit of 32768 is recommended.
?

**Comment by sethrj** [comment]
OK @pcanal , I think I addressed your only comment. @tmdelellis and @stognini, any feedback is welcome and I'll merge tomorrow if it looks good to you.

**Pull request merged**